packages/devextreme/testing/tests/DevExpress.ui.widgets/contextMenu.async.tests.js

```diff
@@ -60,5 +67,59 @@ QUnit.module('Context menu', () => {
 
         assert.strictEqual(instance.option('items').length, 2, 'items.length');
     });
+
+    QUnit.test('Context menu should have correct height on async render (T1258881)', function(assert) {
+        const done = assert.async();
+
+        const menuTargetSelector = '#menuTarget';
+        const items = [{
+            text: 'root',
+            items: [
+                { text: 'sub 1', template: 'myTemplate' },
+                { text: 'sub 2', template: 'myTemplate' },
+                { text: 'sub 3', template: 'myTemplate' },
+                { text: 'sub 4', template: 'myTemplate' },
+                { text: 'sub 5', template: 'myTemplate' },
+                { text: 'sub 6', template: 'myTemplate' },
+                { text: 'sub 7', template: 'myTemplate' },
+            ]
+        }];
+
+        const instance = new ContextMenu($('#simpleMenu'), {
+            target: menuTargetSelector,
+            items,
+            templatesRenderAsynchronously: true,
+            itemTemplate: 'myTemplate',
+            integrationOptions: {
+                templates: {
+                    myTemplate: {
+                        render({ model, container, onRendered }) {
+                            setTimeout(() => {
+                                container.append($('<div>').text(model.text));
+                                onRendered();
+                            });
+                        }
+                    }
+                }
+            },
+        });
+
+        $(menuTargetSelector).trigger($.Event('dxcontextmenu'));
+
+        const $itemsContainer = instance.itemsContainer();
+        const $rootItem = $itemsContainer.find(`.${DX_MENU_ITEM_CLASS}`).eq(0);
+        $($itemsContainer).trigger($.Event('dxhoverstart', { target: $rootItem.get(0) }));
```

Comment:
hover cannot be used on mobile devices. Feel free to do smth like
```javascript
        if(!isDeviceDesktop(assert)) {
            assert.ok(true, 'not actual for mobile devices');
            return;
        }
```

Reply:
added click event for mobile devices